src/scanner/cache/InMemoryCahce.ts

```diff
@@ -26,7 +26,7 @@ export class InMemoryCache implements ICache {
 
   async getOrSet<T>(key: string, setter: () => Promise<T>): Promise<T> {
     const previous = this.get(key);
-    if (previous) {
+    if (previous !== undefined) {
```

Comment:
Because otherwise the case won't be able to store `null`. I mean, if `previous` was `null`, the condition above would be false and the `setter` would be called.